core/trino-main/src/test/java/io/trino/execution/TestRenameCatalogTask.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import io.trino.client.NodeVersion;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.plugin.tpch.TpchPlugin;
+import io.trino.spi.TrinoException;
+import io.trino.spi.resourcegroups.ResourceGroupId;
+import io.trino.sql.tree.CreateCatalog;
+import io.trino.sql.tree.Identifier;
+import io.trino.sql.tree.NodeLocation;
+import io.trino.sql.tree.Property;
+import io.trino.sql.tree.RenameCatalog;
+import io.trino.sql.tree.Statement;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.StandaloneQueryRunner;
+import org.junit.jupiter.api.Test;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static io.trino.execution.querystats.PlanOptimizersStatsCollector.createPlanOptimizersStatsCollector;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.TestingSession.testSession;
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+final class TestRenameCatalogTask
+{
+    private static final List<Property> TPCH_PROPERTIES = ImmutableList.of(
```

Comment:
There's only one use of this. Also, the locations don't make sense unless in the context of the AST nodes this is used. Inline it.